### ISSUE TITLE
Extend Signer interface for status updates

### ISSUE THREAD

**Issue by jacogr** [polkadot-js/api#674]
cc @ianhe8x 

Started playing a bit with the signer, I think it could make sense to make it a bit richer to show the send part to submit the actual status through as well. This way both the dapp and the signer app can actually track what is happening and display status as/when required.

So what I'm basically suggesting -

```js
export interface Signer {
  sign (e: SubmittableExtrinsic, a: string, o: SignatureOptions): Promise<number>;
  update?: (id: number, status: Hash | SubmittableResult) => void 
}
```

First off, the sign returns an internal id that identifies the operation. An optional `update` on the signer interface then can receive status updates from the sending operation -

- `id: number` - the id that was returned via `sign`
- `status: Hash | SubmittableResult` - returns `Hash` when the `signAndSend` has no callback, otherwise the `SubmittableResult`

Basically the status updates gets provided to both the signer as well as the underlying call that is made. So we just check `api.signer && api.signer.update` and then provide the status as required to the signer in addition to the actual caller.

Stumbled across this in `apps` when trying to get the JS console to have the updates, but have the signing done by the apps signer. And both the apps UI as well as the actual JS call needs the ability to have the status.

(As an complete off-topic here, would love to integrate external signers as/when they are available in apps, so if you have something, would love to start working with you on that to integrate over there)

**Comment by jacogr** [comment]
Make a start of how I believe it should work in #675

**Comment by jacogr** [comment]
PS: We could also make ‘update’ non-optional, since it is basically using no info returned, it can just be a noop if the specific signer is not interested in the info.

**Comment by ianhe8x** [comment]
It didn't occur to me, but yes some type of signer will need this to get notified the latest status.

**Comment by ianhe8x** [comment]
> (As an complete off-topic here, would love to integrate external signers as/when they are available in apps, so if you have something, would love to start working with you on that to integrate over there)
The other team is on that. I will forward them your words, and they will be very happy to contact.
